sherpa/plot/utils.py

```diff
@@ -0,0 +1,88 @@
+#
+#  Copyright (C) 2021
+#  Smithsonian Astrophysical Observatory
+#
+#
+#  This program is free software; you can redistribute it and/or modify
+#  it under the terms of the GNU General Public License as published by
+#  the Free Software Foundation; either version 3 of the License, or
+#  (at your option) any later version.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License along
+#  with this program; if not, write to the Free Software Foundation, Inc.,
+#  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+import numpy as np
+
+__all__ = ('intersperse', 'histogram_line')
+
+
+def intersperse(a, b):
+    '''Interleave two arrays a and b
```

Comment:
So this routine makes n assumptions on the ordering? ie `intersperse([1, 7, 3, 4], [100, -200, 34, 5])` will work (even if not really meaningful, and would fail because they're not numpy arrays?)

Reply:
I don't know. This routine was originally written by @DougBurke I just moved it to a different files for (a) easier testing and (b) availability from different plotting backends (it's originally from the pylab_matplotlib, where I shoudl now import this version and remove the definition there. Alas, I forgot...

Reply:
I'd originally put it within another routine, so it was really local to that routine. Here it is now top-level (but not exported) routine. Do we need more-stringent documentation for cases like this?

I was going to mention that we need to remove the code from `pylab_backend`.  I was going to ask whether it's worth splitting this out of the pylab code as we don't use it outside of the pylab code (other than in a test added in this PR). Or am I missing something?

Reply:
We don't currently use it elsewhere. But we will probably need it for any other backend, if we want the plots to look similar. Plus the code is not matplotlib specific at all, so it does not hurt to have it in it's own file with its own tests.

Reply:
I think current lelvel of documentation is sufficient for a very short, not exported helper function.

Reply:
Actually, it is exported (it's listed in `_all__`) . It's still a pretty simple helper routine.

Reply:
For now I'd suggest removing `intersperse` from `__all__`. If needed we can add it later/